thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/metric/transfer/MetricTransfer.java

```diff
@@ -0,0 +1,34 @@
+package com.linkedin.thirdeye.detector.metric.transfer;
+
+import com.linkedin.thirdeye.api.MetricTimeSeries;
+import java.util.List;
+
+
+public class MetricTransfer {
+
+  /**
+   * Use the scaling factor to normalize the input time series
+   *
+   * @param metricsToModify the data MetricTimeSeries to be modified by scaling factor
+   * @param metricName the metricName withing the timeseries to be modified
+   * @param scalingFactorList list of scaling factors
+   * @return
+   */
+  public static void rescaleMetric(MetricTimeSeries metricsToModify, String metricName,
+      List<ScalingFactor> scalingFactorList) {
+    if (scalingFactorList == null || scalingFactorList.size() < 1) {
+      return;  // no transformation if there is no scaling factor in
+    }
+    // Went over the List of Scaling Factors and Rescale the timestamps with in the time range
+    // get the metric name
+    for (long ts : metricsToModify.getTimeWindowSet()) {
```

Comment:
I have thought about this strategy and the reason I am not using the way you suggested is: 1) the MetricTimeSeries do not provide a good way to iterate through each timestamp. Or get the timestamps within a window. 2) If we were to expend the daily factor to 5-minutes interval factor, for example. We need to first align the timestamp (if there is a convention like 5 minutes always start at 0, 5, 10min within an hour).  I agree with you that the way you mentioned here is more graceful. let's discuss further tomorrow morning.